client/landing/stepper/declarative-flow/hosted-site-migration-v2/index.tsx

```diff
@@ -11,14 +18,75 @@ export default {
 	},
 	isSignupFlow: false,
 	useSteps() {
-		return stepsWithRequiredLogin( [ STEPS.IMPORT_OR_MIGRATE_WITH_UPGRADE ] );
+		return stepsWithRequiredLogin( [
+			PLATFORM_IDENTIFICATION,
+			SITE_CREATION_STEP,
+			PROCESSING,
+			SITE_MIGRATION_UPGRADE_PLAN,
+		] );
 	},
 
-	//TEMP: This is a temporary implementation to make the flow work
-	// eslint-disable-next-line @typescript-eslint/no-unused-vars
 	useStepNavigation( currentStep, navigate ) {
+		const [ query ] = useSearchParams();
+
 		return {
-			submit: () => alert( 'Go' ),
+			submit: ( props?: ProvidedDependencies ) => {
+				const data = {
+					platform: props?.platform || query.get( 'platform' ),
+					siteId: props?.siteId || query.get( 'siteId' ),
+					siteSlug: props?.siteSlug || query.get( 'siteSlug' ),
+					next: props?.next || query.get( 'next' ),
+					url: props?.url,
+				} as Record< string, Primitive >;
+
+				if ( currentStep === PLATFORM_IDENTIFICATION.slug ) {
+					return navigate(
+						addQueryArgs(
+							{
+								platform: data.platform,
+								next: data.url,
+							},
+							STEPS.SITE_CREATION_STEP.slug
+						)
+					);
+				}
+
+				if ( currentStep === SITE_CREATION_STEP.slug ) {
+					return navigate(
+						addQueryArgs(
+							{
+								platform: data.platform,
+								next: data.next,
+							},
+							STEPS.PROCESSING.slug
+						)
+					);
+				}
+
+				if ( currentStep === PROCESSING.slug ) {
+					if ( data?.next ) {
+						return window.location.assign(
```

Comment:
I noticed now that in the last version of the import step, we have a _"I want to migrate my entire site"_ link to navigate to the upgrade step.

Reply:
If I understood correctly, it will only happens when the user clicks on `I want to import my content only` on the upgrade screen.

Since this link is still not available, we can add the link `"I want to migrate my entire site" link` as part of the issue 

**Step: Upgrade prompt**
[#93297](https://github.com/Automattic/wp-calypso/issues/93297) 

Does it make sense?

Reply:
> If I understood correctly, it will only happens when the user clicks on I want to import my content only on the upgrade screen.

Yep! Checking the flows, I think that's the only way to navigate to the "Import content from WordPress".

> Since this link is still not available, we can add the link "I want to migrate my entire site" link as part of the issue

Sure! No problem! I added another item to that issue checklist.